assets/section-product-recommendations.css

```
@@ -2,6 +2,10 @@
   display: block;
 }
 
+.product-recommendations:not(.product-recommendations--loaded) {
+  display: none;
```

My previous suggestion actually has some unwanted side-effects. `IntersectionObserver` which is used to trigger the network request doesn't play well with `display: none` elements. 

```suggestion
  padding-bottom: 0;
  padding-top: 0;
```

---

Oh interesting - is that what was happening... I had some inconsistent behaviour but it looked like it was to do with files not updating right away (the `{% javascript %}` tag has some aggressive caching.

Good catch!